src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -347,12 +347,19 @@ protected function instantiateObject(array &$data, string $class, array &$contex
             }
 
             $constructorParameters = $constructor->getParameters();
+            $attributesMetadata = $this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($class)->getAttributesMetadata() : null;
 
             $params = [];
             foreach ($constructorParameters as $constructorParameter) {
                 $paramName = $constructorParameter->name;
                 $key = $this->nameConverter ? $this->nameConverter->normalize($paramName, $class, $format, $context) : $paramName;
 
+                $attributeContext = [];
+
+                if ($attributesMetadata && \array_key_exists($paramName, $attributesMetadata)) {
+                    $attributeContext = $attributesMetadata[$paramName]->getDenormalizationContextForGroups([]);
```

Comment:
Groups are omitted here, is it expected? Why not using `AbstractNormalizer::getGroups()`?

You might actually want to move `AbstractObjectNormalizer::getAttributeDenormalizationContext` method to `AbstractObjectNormalizer` to reuse this logic?

Reply:
No, i didnt know how to handle this right tbh.

Reply:
And i think you meant to move `AbstractObjectNormalizer::getAttributeDenormalizationContext` to `AbstractNormalizer::getAttributeDenormalizationContext`, right?

Reply:
I meant moving this method from `AbstractObjectNormalizer` class to `AbstractNormalizer ` class indeed